File: lib/py/test/thryft_test/protocol/test/json_protocol_test.py
from thryft.protocol.json_protocol import JsonProtocol
from thryft_test.protocol.test._protocol_test import _ProtocolTest


class JsonProtocolTest(_ProtocolTest):
    def _test(self, in_object):
        oprot = JsonProtocol()
        in_object.write(oprot)
        ojson = str(oprot)
        iprot = JsonProtocol(ojson)
        out_object = in_object.read(iprot)
        self.assertEquals(in_object, out_object)

File: lib/cpp/test/thryft_test/protocol/test/json_protocol_test.cpp
#include "./protocol_test.hpp"

#include "thryft/protocol/json_input_protocol.hpp"

namespace thryft_test {
namespace protocol {
namespace test {
using ::thryft::protocol::JsonInputProtocol;

INSTANTIATE_TYPED_TEST_CASE_P(JsonInputProtocol, ProtocolTest, JsonInputProtocol);
}
}
}
